scrape/scrape.go

```diff
@@ -1597,6 +1611,17 @@ loop:
 		meta = metadata.Metadata{}
 		metadataChanged = false
 
+		if sl.labelsDropped {
+			// If we have a scrape with a label that's being dropped then that label should be not be used
+			// for scrape cache. It's possible that dropped label has a different value on every scrape
+			// in which case scrape cache get() would report it as a new sample each time.
+			// That would cause this time series to be reported as being continuesly added to TSDB
+			// when it's not.
+			p.Metric(&lset)
+			lset = sl.sampleMutator(lset)
+			met = []byte(lset.String())
```

Comment:
Maybe it would be better to stop using series string as cache keys and instead use the hash of labels? We would have to hash each series all the time for this. I guess that's why raw response string is being used here instead

Reply:
See https://github.com/prometheus/prometheus/pull/13050
But also, the raw response string should be cheaper to obtain than parsing out the labels.